### PR TITLE
#5498 issues fix pull request for SSIMMetric

### PR DESCRIPTION
**Pull request by Joeycho**
Fixes https://github.com/Project-MONAI/MONAI/issues/5498

### Description

simply add arguments (reduction and get_not_nans) for parent class and update ssim loss function to inherit loss, not nn.module.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Joeycho** [comment]
@wyli Sure, I just added test part.

https://github.com/Project-MONAI/MONAI/pull/5502/commits/a444ea822f7b6431b8054c0c845d7909ab3f2ab2

File: monai/losses/ssim_loss.py [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
 import torch
 import torch.nn.functional as F
-from torch import nn
+from torch.nn.modules.loss import _Loss
 
 from monai.utils.type_conversion import convert_to_dst_type
 
 
-class SSIMLoss(nn.Module):
+class SSIMLoss(_Loss):
```
Comment (wyli) [inline monai/losses/ssim_loss.py:19]: the `forward` is currently hard-coded to return the mean value which is not compatible with the metrics... could you help add the `reduction` option to this class?

Comment (Joeycho) [reply monai/losses/ssim_loss.py:19]: [a2a351db00b91a3bd229a1c546dec48d5d76466f](https://github.com/Project-MONAI/MONAI/pull/5502/commits/a2a351db00b91a3bd229a1c546dec48d5d76466f)

Yes, I think metric should report for all elements in each batch, but SSIMLoss return a single value across in each batch. That was an issue.

Temporarily, I used for-loop and if-else, in the batch, but in long term, I think we should break the dependency between SSIMLoss and SSIM Metric, which is also true in DiceLoss and DiceMetric.

Comment (Joeycho) [reply monai/losses/ssim_loss.py:19]: Oh, so, rather than editing SSIMLoss,  I edited SSIMMetric, since current SSIMLoss makes sense for loss function.

Comment (wyli) [reply monai/losses/ssim_loss.py:19]: makes sense, thanks!

**Review by wyli — APPROVED** [review]
